core/src/main/java/org/apache/iceberg/actions/FileRewritePlanner.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import java.util.Map;
+import java.util.Set;
+import org.apache.iceberg.ContentFile;
+import org.apache.iceberg.ContentScanTask;
+
+/**
+ * A class for planning content file rewrites.
```

Comment:
Planning needs more defining here imho.

The notes below are not clear to me. They seem to be for the entire Operation. I think if we were going to write that there it should be something like

A Rewrite Operation is performed by several classes working in conjunction.

1) A FileRewritePlanner - (this class) which scans the files in the table and determines which files should be rewritten and how they should be grouped. This produces a FileRewritePlan.
2) FileRewritePlan - This immutable output of the planner contains a set of Groups, each containing a set of files that are ment to be compacted together by the FileRewriteRunner.
3) The FileRewrite Runner is the engine specific implementation that can take a FileRewritePlan and perform a rewrite on each of the groups within.

?

Reply:
Fixed, please take a look